commit 8d22f309384cc410da91543f1eb30fe5daf91c3b
Author: Baruch Siach <baruch@tkos.co.il>
Date:   Wed Dec 23 18:43:24 2015 +0200

    i2c: designware: retry transfer on transient failure
    
    Set the i2c_adapter retries field to a sensible value. This allows the i2c core
    to retry master_xfer() when it returns -EAGAIN. Currently the i2c-designware
    driver returns -EAGAIN only on Tx arbitration failure (DW_IC_TX_ARB_LOST).
    
    Reported-by: Rolland Chau <zourongrong@gmail.com>
    Signed-off-by: Baruch Siach <baruch@tkos.co.il>
    Acked-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Signed-off-by: Wolfram Sang <wsa@the-dreams.de>

diff --git a/drivers/i2c/busses/i2c-designware-core.c b/drivers/i2c/busses/i2c-designware-core.c
index 8c48b27..8055cbd 100644
--- a/drivers/i2c/busses/i2c-designware-core.c
+++ b/drivers/i2c/busses/i2c-designware-core.c
@@ -854,6 +854,7 @@ int i2c_dw_probe(struct dw_i2c_dev *dev)
 
 	snprintf(adap->name, sizeof(adap->name),
 		 "Synopsys DesignWare I2C adapter");
+	adap->retries = 3;
 	adap->algo = &i2c_dw_algo;
 	adap->dev.parent = dev->dev;
 	i2c_set_adapdata(adap, dev);
